commit b285d09bded48d2be0b76a3d53c73115e3b936eb
Author: Wolfgang Rohdewald <wolfgang@rohdewald.de>
Date:   Thu Sep 11 11:24:18 2014 +0200

    spread.banana: specific error message about unicode objects
    
    which banana does not handle and a new test for this

diff --git twisted/spread/banana.py twisted/spread/banana.py
index 55fe4e5..c5dbf21 100644
--- twisted/spread/banana.py
+++ twisted/spread/banana.py
@@ -14,6 +14,8 @@ for more details.
 
 import copy, cStringIO, struct
 
+from twisted.python.compat import unicode
+
 from twisted.internet import protocol
 from twisted.persisted import styles
 from twisted.python import log
@@ -332,6 +334,8 @@ class Banana(protocol.Protocol, styles.Ephemeral):
                 int2b128(len(obj), write)
                 write(STRING)
                 write(obj)
+        elif isinstance(obj, unicode):
+            raise BananaError("Banana cannot send unicode objects: %r" % (obj,))
         else:
             raise BananaError("could not send object: %r" % (obj,))
 
diff --git twisted/spread/topfiles/7663.feature twisted/spread/topfiles/7663.feature
new file mode 100644
index 0000000..02e60d0
--- /dev/null
+++ twisted/spread/topfiles/7663.feature
@@ -0,0 +1 @@
+spread.banana now raises a specific error message if user tries to encode a unicode object
diff --git twisted/test/test_banana.py twisted/test/test_banana.py
index 534b400..7ce5212 100644
--- twisted/test/test_banana.py
+++ twisted/test/test_banana.py
@@ -46,6 +46,15 @@ class BananaTestCase(unittest.TestCase):
         assert self.result == 'hello'
 
 
+    def test_unicode(self):
+        """
+        Banana does not support unicode, please implement on a higher level
+        """
+        exc = self.assertRaises(banana.BananaError,
+            self.enc.sendEncoded, u'hello')
+        self.assertIn(b'Banana cannot send unicode objects', str(exc))
+
+
     def test_validDialect(self):
         """
         banana only knows dialects none and pb
